internal/locate/region_request.go

```diff
@@ -142,42 +143,90 @@ type RPCRuntimeStats struct {
 	Consume int64
 }
 
+// RecordRPCRuntimeStats uses to record the rpc count and duration stats.
+func (r *RegionRequestRuntimeStats) RecordRPCRuntimeStats(cmd tikvrpc.CmdType, d time.Duration) {
+	stat, ok := r.RPCStats[cmd]
+	if !ok {
+		r.RPCStats[cmd] = &RPCRuntimeStats{
+			Count:   1,
+			Consume: int64(d),
+		}
+		return
+	}
+	stat.Count++
+	stat.Consume += int64(d)
+}
+
+// RecordRPCErrorStats uses to record the rpc error info and count.
+func (r *RegionRequestRuntimeStats) RecordRPCErrorStats(err string) {
```

Comment:
Yes, we need to collect both rpc error and region error.

Reply:
No. I'm suggesting using different functions for different usages even it collects errors to the same map, to avoid making people confused about what arguments (label or formatted string) to pass to the function `RecordRPCErrorStats`.